src/common/tests_dmg_helpers.c

```diff
@@ -1357,6 +1356,8 @@ dmg_storage_device_list(const char *dmg_config_file, int *ndisks,
 						goto out_json;
 					}
 				}
+			} else {
+				continue;
 			}
```

Comment:
that else condition sounds useless .. you will continue anyway

Reply:
Right and that is why I changed the code to not check NULL but run only if it's not NULL. I am not sure how to satisfy the Coverity in this case. Except I can change the logic to check the specific key value and check for it's NULL return.

Reply:
this will not satisfy the coverity issue that i had mentioned. you do need to revert this change.